src/Cms/Languages.php

```diff
@@ -45,11 +46,17 @@ public function create(array $props)
      */
     public function default()
     {
-        if ($language = $this->findBy('isDefault', true)) {
-            return $language;
-        } else {
-            return $this->first();
+        if ($defaults = $this->filterBy('isDefault', true)) {
+            if ($defaults->count() === 1) {
+                return $defaults->first();
+            }
+
+            if ($defaults->count() > 1) {
```

Comment:
That `if` clause is not necessary – the line will only be reached if the `if` clause above didn't match. So you can just throw the Exception here.

Reply:
Are you sure? I thought `if ($defaults = $this->filterBy('isDefault', true))` might even be entered for empty collections. Or not?

Reply:
You are right, the `filterBy()` method actually always returns the collection object, which is always truthy (even if the collection is empty or the filter didn't find any values).

I think we should change it to something like this then:

```php
$defaults = $this->filterBy('isDefault', true);
if ($defaults && $defaults->isNotEmpty()) {
    if ($defaults->count() === 1) {
        return $defaults->first();
    }

    throw new DuplicateException('You cannot have multiple default languages. Please check your language config files.');
}

return $this->first();
```